include/exadg/incompressible_navier_stokes/user_interface/parameters.cpp

```diff
@@ -559,23 +574,70 @@ bool
 Parameters::viscous_problem() const
 {
   return (equation_type == EquationType::Stokes || equation_type == EquationType::NavierStokes ||
-          use_turbulence_model == true);
+          turbulence_model_data.is_active);
+}
+
+bool
+Parameters::viscous_term_is_linear() const
+{
+  return not viscous_term_is_nonlinear();
+}
+
+bool
+Parameters::viscous_term_is_nonlinear() const
+{
+  return (viscosity_is_variable() &&
+          treatment_of_variable_viscosity == TreatmentOfVariableViscosity::Implicit);
+}
+
+bool
+Parameters::viscosity_is_variable() const
+{
+  return turbulence_model_data.is_active || generalized_newtonian_model_data.is_active;
+}
+
+bool
+Parameters::convective_term_is_treated_implicitly() const
+{
+  if(convective_problem())
+  {
+    if(solver_type == SolverType::Steady ||
+       (solver_type == SolverType::Unsteady &&
+        (treatment_of_convective_term == TreatmentOfConvectiveTerm::Implicit)))
+    {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+bool
+Parameters::nonlinear_viscous_term_is_treated_implicitly() const
+{
+  if(viscous_problem())
+  {
+    if(viscous_term_is_nonlinear() &&
+       treatment_of_variable_viscosity == TreatmentOfVariableViscosity::Implicit)
```

Comment:
The second condition should be removed since it is already part of `viscous_term_is_nonlinear()`.